### PR TITLE
Add flighting capability for common.

### PR DESCRIPTION
**Pull request by iamgusain**
### What
Adding flighting capability to common

### Why
To dynamically change the behavior of common code based on the flight value, which can be provided without code deployment

### How
Adding class CommonFlightManager which is used to set the FlightsProvider for CommonFlights.
This FlightsProvider is used to read the value of the flights. If no flightsProvider is set default Values will be used.

### Testing
Using the flighting capability to control whether or not to use NetworkCapability for checking network connectivity.

Corresponding Broker PR: https://github.com/AzureAD/ad-accounts-for-android/pull/2158

### CODE REVIEW

File: common/src/test/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionServiceTest.java [hunk 1]
```diff
@@ -51,7 +51,7 @@
 @RunWith(RobolectricTestRunner.class)
```
Comment (rpdome) [inline common/src/test/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionServiceTest.java:51]: Add TODO: for API>28 tests.

**Review by rpdome — APPROVED** [review]
I assume the test for NetworkCapabilities will follow in a subsequent PR.

File: common/src/test/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionServiceTest.java (hunk 1 above)
Comment (rpdome) [reply common/src/test/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionServiceTest.java:51]: actually, should this be 23? given that M = 23.

**Review by shahzaibj — APPROVED** [review]
It looks like code similar to what we have in broker repo. We should probably just port that code over to this repo, but it doesn't have to be done immediately.

**Comment by iamgusain** [comment]
> It looks like code similar to what we have in broker repo. We should probably just port that code over to this repo, but it doesn't have to be done immediately.

https://github.com/AzureAD/ad-accounts-for-android/pull/2158

**Comment by iamgusain** [comment]
> I assume the test for NetworkCapabilities will follow in a subsequent PR.

I will include telemetry for the new check in a separate PR.

**Pull request merged**